### ISSUE TITLE
Use of MaterialDesign DialogHost disables Context Menu Column Chooser functionality of DevExpress

### ISSUE THREAD

**Issue by nandhini-sr** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#2982]
### Bug explanation

It is observed that using Material Design DialogHost feature renders the DevExpress Data Grid Column Chooser Context Menu non-functional. To demonstrate the same, I'm sharing a demo application, wherein the left side grid of demo application is the simple DevExpress Data Grid where all the options of Column Chooser Context Menu works. But on the right hand side, due to the usage of Material Design DialogHost, the options are not working as expected. 

Application Built using - .NET Framework, WPF, XAML, C#
Packages - Prism, DevExpress Grid & MaterialDesign Themes

![Code Snippet](https://user-images.githubusercontent.com/49654911/204294315-6cb96b32-90fc-421c-92ca-b313dc1a5dc1.PNG)
![GridDemo View](https://user-images.githubusercontent.com/49654911/204294324-81e62183-c3b4-450b-ac5c-ab0ac60c12cd.PNG)

### Version

4.6.1

**Comment by nandhini-sr** [comment]
Sure, will send the link to demo app repo by tomorrow.

Yes, the error can be reproduced in a free trial of DevExpress.

**Comment by nasonov-andrey** [comment]
Hello!
Did you tried to use style `MaterialDesignEmbeddedDialogHost`? Is the problem the same?

**Comment by Erapchu** [comment]
Hello!
Did you tried to use style `MaterialDesignEmbeddedDialogHost`? Is the problem the same?

**Comment by nandhini-sr** [comment]
@nicolaihenriksen You can find the link to demo application here -
https://github.com/nandhini-sr/DataGridDemo.git

**Comment by nandhini-sr** [comment]
@Erapchu , yes I tried using style MaterialDesignEmbeddedDialogHost, but the problem remains same.

The file can be accessed via link shared in ../views/DataGridEmbedded.xaml and using <views:DataGridEmbedded/> in MainWindow.xaml

**Comment by nicolaihenriksen** [comment]
> You can find the link to demo application here -

@nandhini-sr I may need some guidance in reproducing the issue as well. Both sides seem to work for me?

![ColumnChooser](https://user-images.githubusercontent.com/19572699/204470568-86d13344-8d98-4c15-bf3d-35c3278bccbf.gif)

**Comment by nandhini-sr** [comment]
@nicolaihenriksen The issue is NOT with the checkbox. The issue is when you RIGHT-CLICK on column headers - "Description", "Test Number", "Test facility" INSIDE the context menu, the options like Sort Ascending, Descending, Clear Sorting, Grouping, Show-Hide etc.. DOES NOT WORK. Sharing video for reference-

https://user-images.githubusercontent.com/49654911/204511706-80b76afe-ba5b-4ff0-b558-3d07fa4322c6.mp4

**Comment by nandhini-sr** [comment]
Please let me know if you can reproduce the issue or require more inputs. Thanks for the support.

**Comment by nicolaihenriksen** [comment]
> Please let me know if you can reproduce the above mentioned issue or if anymore inputs are required. Thanks for the support.

Thanks for the video. I see the same issue now. I will see if I can find the root cause.

**Comment by nicolaihenriksen** [comment]
@nandhini-sr I located the root cause, but I am a bit unsure how to go about mitigating the issue. I think I need to involve @Keboo for this.

The root cause is that the `DialogHost` attempts to activate the window it resides in on a `PreviewMouseDown` event:
https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/f7d9a42df36f57efa9521bf13a7701d003567a79/MaterialDesignThemes.Wpf/DialogHost.cs#L788-L795

For this particular scenario, it seems like a bit of a strange behavior since the user is interacting with "another window"; namely a `ContextMenu` opened from inside the `ColumnChooser` dialog.

If I simply comment out the `window.Activate();` call, then the context menu items changing the sorting starts working again; but what have I broken by omitting that `Activate()` call? That is where I will need input from @Keboo.

**Comment by nandhini-sr** [comment]
That's great. Thanks. Let's wait for @Keboo 's reply on this.

**Comment by Keboo** [comment]
First, thank you both for digging into this. I agree with @nicolaihenriksen that does appear to be the root cause. That code in question actually pre-dates when I was working on the library. It was added to address issue #232. So as long as switching to app and coming back still works I think any change we make should be fine.

I have **not** tested this, but I think we should probably just expand the check to the windows child windows as well (I am making the assumption that DevExpress window is being registered as a child window). 
Something like....
```C#
protected override void OnPreviewMouseDown(MouseButtonEventArgs e)
{
    if (Window.GetWindow(this) is { } window && !window.IsActive && window.OwnedWindows.OfType<Window>().All(x => !x.IsActive))
    {
        window.Activate();
    }
    base.OnPreviewMouseDown(e);
}
```

**Comment by nicolaihenriksen** [comment]
> I have **not** tested this, but I think we should probably just expand the check to the windows child windows as well (I am making the assumption that DevExpress window is being registered as a child window).

I have just tested this, and that fixes the issue.
I'll create a PR with the fix.

**Comment by nandhini-sr** [comment]
Hi @nicolaihenriksen @Keboo ,
Brilliant! Thanks for the great support.
May I know the approx. due date when the new version of Material Design will release including these changes?

**Comment by Keboo** [comment]
@nandhini-sr There are preview builds that are released nightly, so if you need the fixes right away you can always grab the latest preview version of the NuGet to try it out. Typically, I will cut full releases every few months.
